### PR TITLE
Very simple streamlit logging

### PR DESCRIPTION
**Pull request by kongzii**

### CODE REVIEW

File: prediction_market_agent_tooling/tools/streamlit_logging.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        x.get_secret_value() for x in logging_settings.free_access_codes
+    ]:
+        logger.info(f"Using free access code: {free_acess_code}.")
+        return LoggedEnum.FREE_ACCESS, None
+
+    # TODO: Because we are initializing APIKeys (based on environment variables) in random places in the code, the user can not provide their own access keys, such as OPENAI_API_KEY.
+    # Doing that would require to change the environment variables in the code, but Streamlit is using 1 thread per session, and the environment variables are shared between threads.
+    # So other users would get the secrets provided by the first user!
+    # Another option is to refactor the code, such that APIKeys are initialized in a single place, somewhere in the agent's initialisation and then, we can initialize them here based on the user's input and just provide them  to the agent.
+    # After that's done, we should also have unique api keys per registered user.
```
[the remark this answers is not in the document]
Comment (kongzii) [reply prediction_market_agent_tooling/tools/streamlit_logging.py:61]: Yes, please create the issue.

File: prediction_market_agent_tooling/tools/streamlit_logging.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            and user.password.get_secret_value() == password.get_secret_value()
+        ):
+            return user
+
+    return None
+
+
+def streamlit_login() -> tuple[LoggedEnum, LoggedUser | None]:
+    logging_settings = LoggingSettings()
+    free_acess_code = st.query_params.get("free_access_code")
```
[the remark this answers is not in the document]
Comment (kongzii) [reply prediction_market_agent_tooling/tools/streamlit_logging.py]: yeah, fixed 😄

File: prediction_market_agent_tooling/tools/streamlit_logging.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    if logging_settings.free_for_everyone:
+        logger.info("Free access for everyone!")
+        return LoggedEnum.FREE_ACCESS, None
+
+    if free_acess_code is not None and free_acess_code in [
```
[the remark this answers is not in the document]
Comment (kongzii) [reply prediction_market_agent_tooling/tools/streamlit_logging.py]: Why not 😄

File: prediction_market_agent_tooling/tools/streamlit_logging.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
+from enum import Enum
+
+import streamlit as st
+from pydantic import BaseModel, SecretStr
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+from prediction_market_agent_tooling.loggers import logger
+
+
+class LoggedUser(BaseModel):
```
[the remark this answers is not in the document]
Comment (kongzii) [reply prediction_market_agent_tooling/tools/streamlit_logging.py]: Sounds better, renamed.

**Review by gabrielfior — APPROVED** [review]

**Pull request merged**